src/app/public/modules/inline-form/inline-form.component.ts

```diff
@@ -0,0 +1,258 @@
+import {
+  ChangeDetectorRef,
+  ChangeDetectionStrategy,
+  Component,
+  EventEmitter,
+  Input,
+  Output,
+  OnInit,
+  OnDestroy,
+  ElementRef,
+  TemplateRef
+} from '@angular/core';
+
+import 'rxjs/add/observable/zip';
+
+import 'rxjs/add/operator/take';
+
+import {
+  Observable
+} from 'rxjs/Observable';
+
+import {
+  SkyAppWindowRef
+} from '@skyux/core';
+
+import {
+  SkyLibResourcesService
+} from '@skyux/i18n';
+
+import {
+  skySlideDissolve
+} from './animations';
+
+import {
+  SkyInlineFormAdapterService
+} from './inline-form-adapter.service';
+
+import {
+  SkyInlineFormCloseArgs,
+  SkyInlineFormButtonLayout,
+  SkyInlineFormConfig,
+  SkyInlineFormButtonConfig
+} from './types';
+
+@Component({
+  selector: 'sky-inline-form',
+  templateUrl: './inline-form.component.html',
+  styleUrls: ['./inline-form.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  animations: [ skySlideDissolve ]
+})
+export class SkyInlineFormComponent implements OnInit, OnDestroy {
+
+  @Input()
+  public config: SkyInlineFormConfig;
+
+  @Input()
+  public template: TemplateRef<any>;
+
+  @Input()
+  public set showForm(value: boolean) {
+    this._showForm = value;
+
+    if (value) {
+      this.skyAppWindowRef.nativeWindow.setTimeout(() => {
+        this.adapter.applyAutofocus(this.elementRef);
+      });
+    }
+  }
+
+  public get showForm() {
+    return this._showForm;
+  }
+
+  @Output()
+  public close = new EventEmitter<SkyInlineFormCloseArgs>();
+
+  public buttons: SkyInlineFormButtonConfig[];
+
+  private _showForm: boolean = false;
+
+  constructor(
+    private adapter: SkyInlineFormAdapterService,
+    private elementRef: ElementRef,
+    private resourcesService: SkyLibResourcesService,
+    private skyAppWindowRef: SkyAppWindowRef,
+    private changeDetectorRef: ChangeDetectorRef
+  ) {}
+
+  public ngOnInit(): void {
+    if (this.isValidCustomConfig(this.config)) {
+      this.buttons = this.getCustomButtons(this.config.buttons);
+    } else {
+      this.getPresetButtons().then((buttons: any) => {
```

Comment:
Why did the buttons type change to `any`?

Reply:
oops. fixed.